### PR TITLE
Fix: user can enter destination without waiting initial balance prompt to finish

### PR DESCRIPTION
**Pull request by andres-moya**
Issue: If say balance after authentication enabled and customer start to enter destination before balance prompt finished he unexpectedly call wrong number (first digits missing). It affect customer experience who used to work with other cards on the market. They do allow blindly enter destination.

Fix: say balance function was modified. Now instead of going through if/then logic and saying balance it accumulate what to say, and then play prompt expecting input. If anything enetered, it stop saying and collect remaining digits. It returns this "early_destination", that is used in the IVR flow to suppress mention above issue.

### CODE REVIEW

File: common/lib/Class.A2Billing.php [hunk 1]
```diff
@@ -143,6 +143,7 @@ class A2Billing
     public $ipaddress;
     public $rate;
     public $destination;
+	public $early_destination='';
```
Comment (areski) [inline common/lib/Class.A2Billing.php:146]: indent ? there is wrong indent in the entire patch. Please try to respect the surrounding code style, as bad as it is now... :)

File: common/lib/Class.A2Billing.php [hunk 2]
```diff
@@ -2083,63 +2088,102 @@ public function fct_say_balance($agi, $credit, $fromvoucher = 0)
...
 
             }
         }
+
+		// now we will play audios of the balance prompt expecting input
+        $entered='';
+
+        // now start saying and get ready to be interrupted
+        foreach ( $say as $item) {
+          if ( is_integer($item) ) {
```
Comment (areski) [inline common/lib/Class.A2Billing.php:2159]: indent with 4 spaces

**Comment by areski** [comment]
Nice job, I like the approach you took! I made few comments, try to improve the code style.

**Comment by andres-moya** [comment]
I can see only two buttons
"Close and comment" and "Comment"
There is no edit button... which one is correct? :)

**Comment by andres-moya** [comment]
awesome.

**Pull request closed without merging**